sdks/java/core/src/test/java/org/apache/beam/sdk/metrics/MetricResultsMatchers.java

```diff
@@ -211,17 +212,21 @@ private static class MatchNameAndKey<T> extends TypeSafeMatcher<MetricResult<T>>
 
     private final String namespace;
     private final String name;
-    private final String step;
+    private final @Nullable String step;
```

Comment:
`MetricKey.create` has parameter type `(@Nullable String, MetricName)`. It already supported null step